lucene/core/src/java/org/apache/lucene/search/FuzzyQuery.java

```diff
@@ -156,9 +159,14 @@ public Automaton toAutomaton() {
 
   @Override
   public void visit(QueryVisitor visitor) {
-    // TODO find some way of consuming Automata
-    if (visitor.acceptField(term.field())) {
-      visitor.visitLeaf(this);
+    if (visitor.acceptField(field)) {
+      if (maxEdits == 0 || prefixLength >= term.text().length()) {
+        visitor.consumeTerms(this, term);
+      } else {
+        CompiledAutomaton a = new CompiledAutomaton(toAutomaton(), null, true,
```

Comment:
May this instating be a little bit expensive?

Reply:
It is, which is why I'm still not convinced this is the correct API.  Our whole setup around automata and utf8 vs utf32 is confusing and hard to use unfortunately